src/services/SyncManager.ts

```
@@ -16,37 +16,51 @@ export class SyncManager<Local> {
 
-    // map rows to Local
+    const remoteRows = remoteRowsData || [];
     const remoteItems = (remoteRows as any[]).map(r => this.cfg.mapRow(r));
-
-    // detect local-only items
     const remoteIds = new Set(remoteItems.map((item: any) => (item as any).id));
-    const localOnly = localItems.filter(item => !remoteIds.has((item as any).id));
 
-    return [...remoteItems, ...localOnly];
+    // 2. Detect local-only items (items in input localItems not present in remoteItems)
+    const localOnlyItems = localItems.filter(item => !remoteIds.has((item as any).id));
```

ESLint

[error] Unexpected any. Specify a different type.

(@typescript-eslint/no-explicit-any)